mlflow/tensorflow/__init__.py

```diff
@@ -554,6 +706,45 @@ def predict(self, data):
             return pandas.DataFrame.from_dict(data=pred_dict)
 
 
+class _TF2ModuleWrapper:
+    def __init__(self, model):
+        self.model = model
+
+    def predict(self, data):
+        import tensorflow
+
+        if isinstance(data, (np.ndarray, list)):
+            data = tensorflow.convert_to_tensor(data)
+        else:
+            raise MlflowException(
+                f"Unsupported input data type: {type(data)}, the input data must be"
+                "numpy array or a list."
+            )
+        result = self.model(data)
+        if isinstance(data, tensorflow.Tensor):
```

Comment:
Should this be `isinstance(result, tensorflow.Tensor)`, or does `isinstance(data, tensorflow.Tensor)` imply that the result is a Tensor?